test-data/pybind11_mypy_demo/src/main.cpp

```diff
@@ -134,14 +140,18 @@ void bind_basics(py::module& basics) {
     .def(py::init<double, double>(), py::arg("x"), py::arg("y"))
     .def("distance_to", py::overload_cast<double, double>(&Point::distance_to, py::const_), py::arg("x"), py::arg("y"))
     .def("distance_to", py::overload_cast<const Point&>(&Point::distance_to, py::const_), py::arg("other"))
-    .def_readwrite("x", &Point::x)
+    .def("as_vector", &Point::as_vector)
+    // Note that the trailing newline is required because the generated docstring
+    // is concatenated to the signature, e.g.:
+    // 'some docstring\n(self: pybind11_mypy_demo.basics.Point) -> float\n'
+    .def_readwrite("x", &Point::x, "some docstring\n")
     .def_property("y",
         [](Point& self){ return self.y; },
         [](Point& self, double value){ self.y = value; }
     )
     .def_property_readonly("length", &Point::length)
     .def_property_readonly_static("x_axis", [](py::object cls){return Point::x_axis;})
-    .def_property_readonly_static("y_axis", [](py::object cls){return Point::y_axis;})
+    .def_property_readonly_static("y_axis", [](py::object cls){return Point::y_axis;}, "another docstring")
```

Comment:
This `Point` example is getting a bit out of hand, or rather, it makes it not so obvious which aspect we are actually trying to test (it distracts with actual "point" semantics).

In my opinion it would be nicer if we could add new test cases that more directly express what we want to test for, which is what I tried to prepare in the issue description #16486. In particular it would be nice to have e.g.

```c++
struct FieldAnnotationTest
{
  int a;
  int b;
  int c;
};

  py::class_<FieldAnnotationTest>(m, "FieldAnnotationTest")
      .def_readwrite("a", &FieldAnnotationTest::a)
      .def_readwrite("b", &FieldAnnotationTest::b, "some docstring")
      .def_property_readonly(
          "c",
          [](const FieldAnnotationTest& x) {
            return x.c;
          },
          "some docstring");
```

because it is also an interesting question how a `def_readwrite` is handled (there was this rejected PEP that suggested to allow adding docstring to fields -- i.e., I was wondering if the `--include-docstring` feature tries to do that or not).

Reply:
I think what you're asking for is a good idea, but I'm not sure if I have the time to commit to doing it, since this is something I'm working on outside of work at the moment.   If all the tests can be placed into main.cpp and we can easily piggy-back what's already there, then it might be straight-forward.  Is that what you were thinking?

Reply:
No worries, I can also take care of that in a follow-up.